opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp

```diff
@@ -184,4 +184,25 @@ data::Aquifers BlackoilAquiferModel<TypeTag>::aquiferData() const
     return data;
 }
 
+template<typename TypeTag>
+template<class Serializer>
+void BlackoilAquiferModel<TypeTag>::
+serializeOp(Serializer& serializer)
+{
+    for (auto& aiPtr : aquifers) {
+        auto* ct = dynamic_cast<AquiferCarterTracy<TypeTag>*>(aiPtr.get());
+        auto* fetp = dynamic_cast<AquiferFetkovich<TypeTag>*>(aiPtr.get());
+        auto* num = dynamic_cast<AquiferNumerical<TypeTag>*>(aiPtr.get());
+        if (ct) {
+            serializer(*ct);
+        } else if (fetp) {
+            serializer(*fetp);
+        } else if (num) {
+            serializer(*num);
+        } else {
+            assert(0);
```

Comment:
While we don't do it right now, I believe there is ongoing work to re-enable `NDEBUG` in Release mode&mdash;i.e., to [disable all `assert()`s](https://github.com/OPM/opm-common/blob/63bfa8e47cbc1742b20a5b513ffcbe10658ca6f1/cmake/Modules/UseOptimization.cmake#L66-L69)  when building in Release mode.  In that case, we probably want more protection than just `assert(false)` here.

Reply:
hmm. i'm not a fan of throwing due to logic/programming errors, but i see you point. exception it is then.